orchestration/dagster_orchestration/hca_manage/verify_release_manifest.py

```diff
@@ -0,0 +1,128 @@
+"""
+Given a file containing a list of constituent staging dirs for a DCP release (aka a manifest),
+verify that data has been loaded from each of them to the target DCP dataset and the count of loaded files
+matches the # in the staging area
+
+All datarepo load totals are inferred from a supplied start date for the import process.
+
+Example invocation:
+python verify_release_manifest.py -s 2021-03-24 -f testing.csv -g fake-gs-project -b fake-bq-project -d fake-dataset
+"""
+import argparse
+import logging
+import sys
+from urllib.parse import urlparse
+
+from google.cloud import bigquery, storage
+from hca_orchestration.contrib import google as hca_google
+
+logging.basicConfig(level=logging.INFO, format='%(message)s')
+
+
+def get_expected_load_totals(storage_client, staging_areas):
+    """
+    Given a list of GS staging areas, count the files present in each /data subdir
+    """
+    expected = {}
+    for staging_area in staging_areas:
+        url = urlparse(staging_area)
+        prefix = f"{url.path.lstrip('/')}/data/"
+        blobs = list(storage_client.list_blobs(url.netloc, prefix=prefix))
+        expected[staging_area] = len(blobs)
+    return expected
+
+
+def get_load_history(bq_project, dataset, start_date):
+    client = bigquery.Client(project=bq_project)
+
+    # Determine the number of distinct files loaded after the given start date, grouped by staging area
+    query = f"""
+                WITH base as (
+                    SELECT distinct(source_name)
+                    FROM `datarepo_{dataset}.datarepo_load_history` dlh
+                    LEFT JOIN `datarepo_{dataset}.sequence_file` sf
+                    ON sf.file_id = dlh.file_id
+                    LEFT JOIN `broad-datarepo-terra-prod-hca2.datarepo_hca_prod_20201120_dcp2.analysis_file` af
```

Comment:
Do we want to keep the project and dataset names hardcoded here? I don't think there is a strong argument at the moment for pulling them out, but it could be helpful in the future esp. considering the great migration to real prod in the repo

Reply:
I went to the trouble of not hardcoding them in the `FROM` clause above, they should be unhardcoded here as well 👍